templates/triton_services/README.md

```diff
@@ -0,0 +1,175 @@
+and [Triton Ray Serve Deployment](https://github.com/triton-inference-server/tutorials/tree/r24.04/Triton_Inference_Server_Python_API/examples/rayserve).
+If you need to learn more about TensorRT and Triton, refer to their official
+documentation and tutorial.
+
+In Nvidia's tutorial, they have an image to build the model and another image to serve
+the model. In this tutorial, the docker image pulls in all the necessary dependencies.
+This image compiles the model, allow you to do local development in Anyscale Workspace,
+and deploys to Anyscale Services.
+
+The full `Dockerfile` used to build the image is already provided in the workspace.
+You can view the file by clicking on the `Dockerfile` in the file explorer. It shows
+what's required in the image to run this tutorial.
+
+## Compile model using Triton's Python backend on Anyscale Workspaces
+
+Unlike Nvidia's tutorial, this tutorial doesn't include models in the Docker image.
+Instead, in this section, you are building the model and upload the model to a cloud
+storage such as AWS S3 or GCP Storage for serving later.
+
+Run this code to start a Triton server using the `/tmp/workspace/diffusion-models`
+directory as the model repository.
+
+
+```python
+import tritonserver
+
+model_repository = ["/tmp/workspace/diffusion-models"]
+
+triton_server = tritonserver.Server(
+    model_repository=model_repository,
+    model_control_mode=tritonserver.ModelControlMode.EXPLICIT,
+)
+triton_server.start(wait_until_ready=True)
+```
+
+Compile the model using Triton's Python backend. This takes 10-15 minutes on a
+T4 GPU and 8-10 minutes on an A10G GPU. The model saves in the `model_repository`
+directory as TensorRT engine artifacts. Also keep in mind the model compiling
+needs to in the same type of GPU you are planning to serve the model on.
+
+
+
+```python
+import time
+import datetime
+
+
+print(f"start time: {datetime.datetime.now()}")
+t0 = time.time()
+model = triton_server.load("stable_diffusion_1_5")
+duration = time.time() - t0
+print(f"Total duration: {duration}s")
+
+# Unload the model and the server to free the memory.
+triton_server.unload(model, wait_until_unloaded=True)
+triton_server.stop()
+```
+
+After the model compile, you can upload the model to a cloud storage. You need to
+upload both the model config file `config.pbtxt` and the TensorRT engine model
+directory. Anyscale provides a environment variable `ANYSCALE_ARTIFACT_STORAGE` for
+customers to store model artifacts. Use one of the following to upload the model.
+
+
+```python
+# If you are running in AWS.
+aws s3 cp /tmp/workspace/diffusion-models/stable_diffusion_1_5/config.pbtxt $ANYSCALE_ARTIFACT_STORAGE/triton_model_repository/stable_diffusion_1_5/config.pbtxt
+aws s3 cp /tmp/workspace/diffusion-models/stable_diffusion_1_5/1/1.5-engine-batch-size-1/ $ANYSCALE_ARTIFACT_STORAGE/triton_model_repository/stable_diffusion_1_5/1/1.5-engine-batch-size-1/ --recursive
```

Comment:
seems like the first copy of the text file isn't necessary if you just copy the whole `triton_model_repository/stable_diffusion_1_5/` directory

Reply:
True, but there are also some .py and onnx files getting autogenerated in this directory. I don't want to confuse the user to copy those other files that's not needed. Also it helps to save time not to upload the entire directory :)

Reply:
Got it. Maybe specify that those aren't needed in a comment